model/minio/minio_server.rb

```diff
@@ -79,11 +75,7 @@ def check_pulse(session:, previous_pulse:)
     pulse
   end
 
-  def dns_zone
-    @dns_zone ||= DnsZone.where(project_id: Config.minio_service_project_id, name: Config.minio_host_name).first
-  end
-
-  def url
-    dns_zone ? "http://#{cluster.hostname}:9000" : connection_string
+  def server_url
+    cluster.dns_zone ? cluster.url : ip4_url
```

Comment:
IMHO, the server shouldn't check cluster's dns_zone existence. 
We may move this check to cluster's `def url` or `def hostname`
```ruby
  def url
      dns_zone ? "http://#{hostname}:9000" : nil
  end
```
Then
```suggestion
    cluster.url || ip4_url
```